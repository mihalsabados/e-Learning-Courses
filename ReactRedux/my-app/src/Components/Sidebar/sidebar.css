
.left-sidebar{
    margin-top: 20px;
    width: 27%;
}

.navigation-header, .contact-header{
    height: 25px;
    background-color: #d8d8d8;
}

.navigation-header p, .contact-header p{
    color: #3a434b;
    font-weight: 700;
    font-size: 14px;
    margin-left: 10px;
    padding-top: 5px;
}

.navigation-content{
    margin-block: 22px;
}

.navigation-item{
    height: 80px;
    margin-bottom: 10px;
    background-color: white;
    display: flex;
}

.nav-item-img, .nav-item-img img{
    width: 70px;
    margin-left: 5px;
    margin-top: 2px;
}

.nav-item-text{
    margin-left: 20px;
    margin-right: 30px;
    display: flex;
    align-items: center;
}

.nav-item-text{
    font-weight: 600;
    font-size: 17px;
}

.contact-item{
    margin-left: 30px;
    height: 60px;
    margin-top: 25px;
    font-size: 13px;
}

.contact-item p{
    margin-block: 5px;
    color: rgb(63, 63, 63);
}

.contact-item a{
    text-decoration: none;
}

.c-type{
    font-weight: 600;
}