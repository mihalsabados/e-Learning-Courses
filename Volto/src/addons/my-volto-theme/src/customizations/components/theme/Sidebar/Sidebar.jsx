function Sidebar() {
  return (
    <div className="left-sidebar">
      <div className="navigation-header">
        <p>Navigation</p>
      </div>
      <div className="navigation-content">
        <div className="navigation-item">
          <div className="nav-item-img">
            <img src="resources/fraud_prevention.png" alt="Fraud Prevention" />
          </div>
          <div className="nav-item-text">
            <p>Money Laundering and Fraud Prevention</p>
          </div>
        </div>
        <div className="navigation-item">
          <div className="nav-item-img">
            <img src="resources/it_security.png" alt="It Security" />
          </div>
          <div className="nav-item-text">
            <p>IT security and information security</p>
          </div>
        </div>
        <div className="navigation-item">
          <div className="nav-item-img">
            <img src="resources/data_protection.png" alt="Data Protection" />
          </div>
          <div className="nav-item-text">
            <p>Data Protection</p>
          </div>
        </div>
        <div className="navigation-item">
          <div className="nav-item-img">
            <img src="resources/wphg.png" alt="WpHG" />
          </div>
          <div className="nav-item-text">
            <p>WpHG-Compliance</p>
          </div>
        </div>
      </div>

      <div className="contact-header">
        <p>Contact</p>
      </div>
      <div className="contact-content">
        <div className="contact-item">
          <p className="c-type">Technical support</p>
          <p>John Doe</p>
          <a href="mailto: it@example.com">it@example.com</a>
        </div>
        <div className="contact-item">
          <p className="c-type">Content-didactic topics</p>
          <p>Jane Roe</p>
          <a href="mailto: content@example.com">content@example.com</a>
        </div>
      </div>
    </div>
  );
}

export default Sidebar;
