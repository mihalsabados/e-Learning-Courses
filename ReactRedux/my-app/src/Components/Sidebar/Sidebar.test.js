/* eslint-disable no-undef */
import React from "react";
import { render, screen } from "@testing-library/react";
import Sidebar from "./Sidebar";
import renderer from "react-test-renderer";

test("Sidebar component should render", async () => {
	render(<Sidebar />);

	// Assert that the navigation header is present
	expect(screen.getByText("Navigation")).toBeInTheDocument();

	// Assert that each navigation item is present
	expect(screen.getByAltText("Fraud Prevention")).toBeInTheDocument();
	expect(
		screen.getByText("Money Laundering and Fraud Prevention")
	).toBeInTheDocument();

	expect(screen.getByAltText("It Security")).toBeInTheDocument();
	expect(
		screen.getByText("IT security and information security")
	).toBeInTheDocument();

	expect(screen.getByAltText("Data Protection")).toBeInTheDocument();
	expect(screen.getByText("Data Protection")).toBeInTheDocument();

	expect(screen.getByAltText("WpHG")).toBeInTheDocument();
	expect(screen.getByText("WpHG-Compliance")).toBeInTheDocument();

	// Assert that the contact header is present
	expect(screen.getByText("Contact")).toBeInTheDocument();

	// Assert that each contact item is present
	expect(screen.getByText("Technical support")).toBeInTheDocument();
	expect(screen.getByText("John Doe")).toBeInTheDocument();
	expect(screen.getByText("it@example.com")).toBeInTheDocument();

	expect(screen.getByText("Content-didactic topics")).toBeInTheDocument();
	expect(screen.getByText("Jane Roe")).toBeInTheDocument();
	expect(screen.getByText("content@example.com")).toBeInTheDocument();
});

test("Email links in Sidebar component should have right href", () => {
	render(<Sidebar />);

	expect(screen.getByRole("link", { name: "it@example.com" })).toHaveAttribute(
		"href",
		"mailto: it@example.com"
	);
	expect(
		screen.getByRole("link", { name: "content@example.com" })
	).toHaveAttribute("href", "mailto: content@example.com");
});

test("Sidebar component snapshot", () => {
	const component = renderer.create(<Sidebar />);
	const tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
