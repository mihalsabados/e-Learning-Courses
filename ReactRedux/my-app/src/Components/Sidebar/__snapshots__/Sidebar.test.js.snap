// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component snapshot 1`] = `
<div
  className="left-sidebar"
>
  <div
    className="navigation-header"
  >
    <p>
      Navigation
    </p>
  </div>
  <div
    className="navigation-content"
  >
    <div
      className="navigation-item"
    >
      <div
        className="nav-item-img"
      >
        <img
          alt="Fraud Prevention"
          src="resources/fraud_prevention.png"
        />
      </div>
      <div
        className="nav-item-text"
      >
        <p>
          Money Laundering and Fraud Prevention
        </p>
      </div>
    </div>
    <div
      className="navigation-item"
    >
      <div
        className="nav-item-img"
      >
        <img
          alt="It Security"
          src="resources/it_security.png"
        />
      </div>
      <div
        className="nav-item-text"
      >
        <p>
          IT security and information security
        </p>
      </div>
    </div>
    <div
      className="navigation-item"
    >
      <div
        className="nav-item-img"
      >
        <img
          alt="Data Protection"
          src="resources/data_protection.png"
        />
      </div>
      <div
        className="nav-item-text"
      >
        <p>
          Data Protection
        </p>
      </div>
    </div>
    <div
      className="navigation-item"
    >
      <div
        className="nav-item-img"
      >
        <img
          alt="WpHG"
          src="resources/wphg.png"
        />
      </div>
      <div
        className="nav-item-text"
      >
        <p>
          WpHG-Compliance
        </p>
      </div>
    </div>
  </div>
  <div
    className="contact-header"
  >
    <p>
      Contact
    </p>
  </div>
  <div
    className="contact-content"
  >
    <div
      className="contact-item"
    >
      <p
        className="c-type"
      >
        Technical support
      </p>
      <p>
        John Doe
      </p>
      <a
        href="mailto: it@example.com"
      >
        it@example.com
      </a>
    </div>
    <div
      className="contact-item"
    >
      <p
        className="c-type"
      >
        Content-didactic topics
      </p>
      <p>
        Jane Roe
      </p>
      <a
        href="mailto: content@example.com"
      >
        content@example.com
      </a>
    </div>
  </div>
</div>
`;
